VEDA_OS01/utils.py

```diff
@@ -57,3 +57,53 @@ def invalidate_fernet_cached_properties(model, fields):
             del field.fernet
         except AttributeError:
             pass
+
+
+def get_incomplete_encodes(edx_id):
+    """
+    Get incomplete encodes for the given video.
+
+    Arguments:
+        edx_id: an ID identifying the VEDA video.
+    """
+    encode_list = []
+    try:
+        video = Video.objects.filter(edx_id=edx_id).latest()
+    except Video.DoesNotExist:
+        return encode_list
+
+    course = video.inst_class
+    # Pick the encodes map from the settings.
+    encodes_map = get_config().get('encode_dict', {})
+    # Active encodes according to course instance.
+    for attr, encodes in encodes_map.iteritems():
+        if getattr(course, attr, False):
+            encode_list += [encode.strip() for encode in encodes]
+
+    # Filter active encodes further according to their corresponding encode profiles activation.
+    for encode in list(encode_list):
```

Comment:
`encode_list` is already list then why `list(encode_list)`?

Reply:
to make a copy of it.